Fix authentication

lavonHunter opened the pull request:
support shraga authentication

liadperetz committed:
fix(SignInSecret): now successfuly use it. now have a new problem.

fixed the problem of undefined SignInSecret. Now its getting into
infinite loop.

liadperetz committed:
fix(infinity loop): now its show the board

fixed the infinity loop. now its working for specific user

liadperetz committed:
fix(authentication): now using shraga

now supporting all the users

resolve #58

liadperetz committed:
delete useless prints

liadperetz committed:
refactor(authentication): pushes ADFS profile to users collection

pushes the profile returned by ADFS to the users collection instead of
the old structure. SIP

liadperetz committed:
test new syntax

liadperetz committed:
feat(authentication):updates profile in DB when user logs in.

updates profile in DB when user logs in. takes care of cases where old
structure of profile in db.

resolve #59 #60

liadperetz committed:
refactor(authentication): changes configuration.

adds shragaURL and callbackURL to .env file. updates authConfig and uses
it in passport.js

ShragaUser committed:
Delete launch.json

ShragaUser commented on src/app/components/utils.js:
```diff
@@ -14,3 +14,14 @@ export const colorsWithLabels = [
+    name: profile.name.lastName + " " + profile.name.firstName,
+    mail: profile.mail,
+    display: profile.displayName,
+    imageUrl: null
```
shraga exports imageUrl when available. 
the property is "photo".
do imageUrl: profile.photo || null

ShragaUser commented on src/server/passport.js:
```diff
@@ -17,8 +16,11 @@ const configurePassport = db => {
-      cb(null, user);
+    users.findOne({_id: id }).then(user => {
+      if(user.provider != "ADFS"){
+        cb(null, false);
```
WRONG!
provider can be different than ADFS. 
it can be user-service and is allowed.

ShragaUser commented on src/server/passport.js:
```diff
@@ -43,37 +45,24 @@ const configurePassport = db => {
-              .then(() => done(null, newUser));
-          });
+
+  const config = authConfig();
```
deconstruct for better readability.

ShragaUser commented on src/server/passport.js:
```diff
@@ -43,37 +45,24 @@ const configurePassport = db => {
+      .then(result => {
+        const { matchedCount, modifiedCount } = result;
+        if (matchedCount && modifiedCount) {
+          console.log(`Successfully added a new review.`)
```
review?

ShragaUser commented on src/server/routes/api.js:
```diff
@@ -116,6 +117,8 @@ const api = db => {
 
   router.post("/userId", (req, res) => {
+    console.log("dror");
+    console.log(req.body);
```
delete from PR

ShragaUser commented on src/server/routes/api.js:
```diff
@@ -6,6 +6,7 @@ import {
   PUBLIC_USER_PROPERTIES
 } from "../../constants";
 import { pick } from "../helper";
+import { transformUser } from "../../app/components/utils";
```
imported but never used.

ShragaUser reviewed (changes requested):
left a couple of changes that need addressing. 
also: remember to remove completely local strategy but also allow for multiple users in local idp.

ShragaUser commented:
Also: remember to update .env example in README.md

liadperetz committed:
fix(authentication): fixes bugs

fixes bugs in passport.js
removes logs.

resolve #69

ShragaUser reviewed (changes requested):
Add .env variables to readme.
Delete in /auth the local endpoint.

liadperetz committed:
fix(authentication): fixes .env

adds SHRAGA_URL and CALLBACK_URL to .env.
updates README.md

resolve #69

liadperetz committed:
fix(authentication): creates welcome board for new users.

creates welcome board for new users.
deletes localStrategy from passport.js

resolve #69
